ui/apps/platform/src/hooks/useFetchClustersForPermissions.ts

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect } from 'react';
+
+import { ClusterForPermissions, getClustersForPermissions } from 'services/RolesService';
+import { Cluster } from 'types/cluster.proto';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+
+type Result = {
+    isLoading: boolean;
+    clusters: Cluster[];
+    error: string;
+};
+
+function useFetchClustersForPermissions(permissions: string[]): Result {
+    const defaultResultState = {
+        clusters: [],
+        error: '',
+        isLoading: true,
+    };
```

Comment:
Same comment about default of true for loading here, as I had in the file `ui/apps/platform/src/hooks/useFetchClusterNamespacesForPermissions.ts` above.